### PR TITLE
Role,User, repositories, service classes implemented, SpringBoot conf…

### PR DESCRIPTION
**Pull request by semyonich**
…igured

### CODE REVIEW

File: src/main/java/com/pet/bankservice/service/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+
+    @Override
+    public User save(User user) {
```
Comment (Artificial-Friend) [inline src/main/java/com/pet/bankservice/service/impl/UserServiceImpl.java:16]: as far as I know
`update()` uses `save()` method from crud interface but I see that your intention is reusing `save()` method from service. Either you just forgot it or doing wrong - you need `update()` method in service

File: src/main/java/com/pet/bankservice/repository/UserRepository.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import com.pet.bankservice.entity.User;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+import java.util.Optional;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {
+    @Query("FROM User u INNER JOIN FETCH u.roles WHERE u.id=?1")
+    Optional<User> get(Long aLong);
```
Comment (Artificial-Friend) [inline src/main/java/com/pet/bankservice/repository/UserRepository.java]: Entity field is `id` so why it is `aLong` here?

File: src/main/java/com/pet/bankservice/entity/Role.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    public enum RoleName {
+        ADMIN,
+        USER;
+    }
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Enumerated(EnumType.STRING)
+    private RoleName roleName;
```
Comment (Artificial-Friend) [inline src/main/java/com/pet/bankservice/entity/Role.java:27]: Make it unique, you don't need 15 same ADMIN's

File: src/main/java/com/pet/bankservice/util/InitialDataInject.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.HashSet;
+import java.util.Set;
+import org.springframework.stereotype.Component;
+
+@Component
+public class InitialDataInject {
+    private final UserService userService;
+    private final RoleService roleService;
+
+    public InitialDataInject(UserService userService, RoleService roleService) {
```
Comment (Artificial-Friend) [inline src/main/java/com/pet/bankservice/util/InitialDataInject.java]: I am not sure that you should push that code. Anyway be consistent and use `@AllArgsConstructor` here as well.

File: pom.xml [hunk 5]
```diff
@@ -21,15 +21,10 @@
...
-            <artifactId>spring-boot-starter-thymeleaf</artifactId>
-        </dependency>
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
-
         <dependency>
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-devtools</artifactId>
```
Comment (Artificial-Friend) [inline pom.xml]: Are you sure you need that?

**Review by Artificial-Friend — APPROVED** [review]

File: src/main/java/com/pet/bankservice/service/impl/UserServiceImpl.java (hunk 1 above)
Comment (semyonich) [reply src/main/java/com/pet/bankservice/service/impl/UserServiceImpl.java:16]: Task: 7. You can have a single save() method on the service layer for both ‘create’ and ‘update’ operations.

File: src/main/java/com/pet/bankservice/repository/UserRepository.java (hunk 2 above)
Comment (semyonich) [reply src/main/java/com/pet/bankservice/repository/UserRepository.java]: done

File: src/main/java/com/pet/bankservice/entity/Role.java (hunk 3 above)
Comment (semyonich) [reply src/main/java/com/pet/bankservice/entity/Role.java:27]: done

File: src/main/java/com/pet/bankservice/util/InitialDataInject.java (hunk 4 above)
Comment (semyonich) [reply src/main/java/com/pet/bankservice/util/InitialDataInject.java]: done

File: pom.xml (hunk 5 above)
Comment (semyonich) [reply pom.xml]: done

File: src/main/java/com/pet/bankservice/repository/UserRepository.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.Optional;
+
+@Repository
+public interface UserRepository extends JpaRepository<User, Long> {
+    @Query("FROM User u INNER JOIN FETCH u.roles WHERE u.id=?1")
+    Optional<User> get(Long id);
+
+    @Query("FROM User u INNER JOIN FETCH u.roles WHERE u.phoneNumber=?1")
+    Optional<User> findByPhoneNumber(String phoneNumber);
+
```
Comment (boroda4436) [inline src/main/java/com/pet/bankservice/repository/UserRepository.java]: redundant empty lines

**Review by boroda4436 — APPROVED** [review]

File: src/main/java/com/pet/bankservice/repository/UserRepository.java (hunk 6 above)
Comment (semyonich) [reply src/main/java/com/pet/bankservice/repository/UserRepository.java]: done

**Pull request merged**